File: ready2road/src/app/model/Messaggio.ts
import {Conversazione} from "./Conversazione";
import {ConversazioneService} from "../services/conversazione.service";
import {Inject} from "@angular/core";

export class Messaggio{
  private testo:string;
  private mittente:string;
  private id:number;
  private data:Date;
  private conversazione:Conversazione;

  constructor(json:any, @Inject(ConversazioneService) private service: ConversazioneService) {
    this.testo=json.testo;
    this.mittente=json.mittente;
    this.id=json.id;
    this.data=json.data;
    this.conversazione=new Conversazione(json.conversazione, service);
  }
  getTesto= (): string =>{
    return this.testo;
  }
  getMittente= (): string =>{
    return this.mittente;
  }
  getId= (): number =>{
    return this.id;
  }
  getData= (): Date =>{
    return this.data;
  }
  getConversazione= (): Conversazione=>{
    return this.conversazione;
  }
  setTesto= (testo:string): void =>{
    this.testo=testo;
  }
  setMittente= (mittente:string): void =>{
    this.mittente=mittente;
  }
  setId= (id:number): void =>{
    this.id=id;
  }
  setData= (data:Date): void =>{
    this.data=data;
  }
  setConversazione= (conversazione:Conversazione): void =>{
    this.conversazione=conversazione;
  }

}

File: Ready2RoadBackend/src/main/java/Ready2road/Persistenza/Model/Messaggio.java
package Ready2road.Persistenza.Model;

import com.google.gson.JsonObject;

import java.time.LocalDateTime;

public class Messaggio {
    private String testo;
    private String mittente;
    private Long id;
    private LocalDateTime data;
    private Conversazione conversazione;

    public String getTesto() {
        return testo;
    }

    public void setTesto(String testo) {
        this.testo = testo;
    }

    public String getMittente() {
        return mittente;
    }

    public void setMittente(String mittente) {
        this.mittente = mittente;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDateTime getData() {
        return data;
    }

    public void setData(LocalDateTime data) {
        this.data = data;
    }

    public Conversazione getConversazione() {
        return conversazione;
    }

    public void setConversazione(Conversazione conversazione) {
        this.conversazione = conversazione;
    }

    public JsonObject toJson() {
        JsonObject jsonObject = new JsonObject();
        jsonObject.addProperty("testo", testo);
        jsonObject.addProperty("mittente", mittente);
        jsonObject.addProperty("id", id);
        jsonObject.addProperty("data", data.toString());
        jsonObject.add("conversazione", conversazione.toJson());
        return jsonObject;

    }
}
